### PR TITLE
Add `daita` as a Cargo cfg variable

### PR DESCRIPTION
**Pull request by Serock3**
Due to reasons outlined here https://github.com/mullvad/mullvadvpn-app/pull/6202#pullrequestreview-2042382688, It may be problematic to gate DAITA behind a cargo feature. This is an alternative approach which avoids the use of features.

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)

### CODE REVIEW

File: .github/workflows/cargo-vendor.yml [hunk 1]
```diff
@@ -18,6 +18,11 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Checkout submodules
+        run: |
+          git config --global --add safe.directory '*'
+          git submodule update --init --depth=1 dist-assets/binaries wireguard-go-rs/libwg
```
Comment (hulthe) [inline .github/workflows/cargo-vendor.yml:24]: shouldn't this be wireguard-go-rs/libwg/**wireguard-go**?

File: wireguard-go-rs/build.rs [hunk 2]
```diff
@@ -15,6 +15,8 @@ fn main() {
         "android" => {
             cmd.arg("--android");
         }
+        // TODO: Is is allowed to panic? Won't this always happen if we run `cargo build` in the
+        // workspace root for e.g. windows?
```
Comment (hulthe) [inline wireguard-go-rs/build.rs:19]: ideally I think we shouldn't include wireguard-go-rs as a dependency on windows

Comment (hulthe) [reply wireguard-go-rs/build.rs:19]: but if that's not possible then we should remove the panic, yes

File: .github/workflows/cargo-vendor.yml (hunk 1 above)
Comment (Serock3) [reply .github/workflows/cargo-vendor.yml:24]: woops

File: wireguard-go-rs/build.rs (hunk 2 above)
Comment (Serock3) [reply wireguard-go-rs/build.rs]: Don't think it's possible no. We don't have it as a dep for the daemon, but it's still compiled when just running `cargo build` in the workspace root. I removed the panic.

**Review by MarkusPettersson98 — COMMENTED** [review]
> Reviewed 29 of 53 files at r1, 23 of 33 files at r3, 8 of 10 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)* status: 61 of 64 files reviewed, 3 unresolved discussions (waiting on @hulthe and @Serock3)

___
*[`build.sh` line 168 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxlKDkA3mqdKG4mdv1C:-NxlKDkA3mqdKG4mdv1D:b-am0cls) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6856cc5752377311ae183e65457f71ea2839fd2c/build.sh#L168)):*
> ```Shell
> if [[ "$(uname -s)" == "MINGW"* || "$(uname -s)" == "Linux" ]]; then
>     CARGO_ARGS+=(--features daita)
> fi
> ```

This should be removed, otherwise `build.sh` will fail when compiling due to the daita feature not existing on Linux and Windows :blush:



_Code quote:_
```Shell
# Enable DAITA on supported platforms
if [[ "$(uname -s)" == "MINGW"* || "$(uname -s)" == "Linux" ]]; then
    CARGO_ARGS+=(--features daita)
fi
```

**Review by hulthe — COMMENTED** [review]
> Reviewed 6 of 7 files at r2, 2 of 33 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)* status: 63 of 64 files reviewed, 3 unresolved discussions (waiting on @Serock3)

___
*[`.github/workflows/cargo-vendor.yml` line 24 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxMYKB0BT0No4ckYtjA:-NxMYKB0BT0No4ckYtjB:bpm3v1e) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/359aa96be5e78c84ea4c96c1a6e00b88106aac10/.github/workflows/cargo-vendor.yml#L24)):*
> ```YAML
>         run: |
>           git config --global --add safe.directory '*'
>           git submodule update --init --depth=1 dist-assets/binaries wireguard-go-rs/libwg
> ```

shouldn't this be wireguard-go-rs/libwg/**wireguard-go**?
___
*[`.github/workflows/desktop-e2e.yml` line 164 at r5](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxlgbXT3fhpjBizNm4M:-NxlgbXT3fhpjBizNm4N:b-nnu9au) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/d5be080ce1d6e6042b29dcfa96f1eb650a0bd070/.github/workflows/desktop-e2e.yml#L164)):*
> ```YAML
>         uses: actions/checkout@v2
>       - name: Checkout submodules
>        # TODO: should this specify a path? We don't want to pull `wireguard-go`
> ```

better to have it and not need it? :shrug: 
it's probably a non-issue to just keep this

**Review by dlon — CHANGES_REQUESTED** [review]
> Reviewed 1 of 10 files at r4, 11 of 13 files at r6, 5 of 16 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)* status: 66 of 77 files reviewed, 14 unresolved discussions (waiting on @hulthe, @MarkusPettersson98, and @Serock3)

___
*[`.github/workflows/android-app.yml` line 126 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsnjvgCWSxm-3nQRzX:-NxsnjvgCWSxm-3nQRzY:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L126)):*
> ```YAML
>         run: |
>           git config --global --add safe.directory $(pwd)
>           git submodule update --init wireguard-go-rs/libwg/wireguard-go
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/android-app.yml` line 186 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsoG9bBZ2go4Sw0xak:-NxsoG9bBZ2go4Sw0xal:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L186)):*
> ```YAML
> 
>       - name: Checkout wireguard-go submodule
>         run: git submodule update --init wireguard-go-rs/libwg/wireguard-go
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/android-app.yml` line 245 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsoatN-phMk8DzR9-b:-NxsoatN-phMk8DzR9-c:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L245)):*
> ```YAML
>         uses: actions/checkout@v4
> 
>       - name: Checkout wireguard-go submodule
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/android-app.yml` line 277 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-Nxsoct55BGIosunA_PN:-Nxsoct55BGIosunA_PO:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L277)):*
> ```YAML
>         uses: actions/checkout@v4
> 
>       - name: Checkout wireguard-go submodule
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/android-app.yml` line 386 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsodgL8kfAkVnuLOYg:-NxsodgL8kfAkVnuLOYh:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L386)):*
> ```YAML
>         uses: actions/checkout@v4
> 
>       - name: Checkout wireguard-go submodule
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/cargo-vendor.yml` line 21 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-Nxsoqw03tAOSG_78dIz:-Nxsoqw03tAOSG_78dJ-:b-cs1n92) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/cargo-vendor.yml#L21)):*
> ```YAML
>         uses: actions/checkout@v3
> 
>       - name: Checkout submodules
> ```

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
___
*[`.github/workflows/clippy.yml` line 46 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspVf-4Yb-KWHjDED2:-NxspVf-4Yb-KWHjDED3:b2nreh6) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L46)):*
> ```YAML
>           repo-token: ${{ secrets.GITHUB_TOKEN }}
> 
>       - name: Checkout submodules
> ```

Same as above.
___
*[`.github/workflows/clippy.yml` line 62 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspCtq7F_FDkq7Jt63:-NxspCtq7F_FDkq7Jt64:bnh4ekj) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L62)):*
> ```YAML
> 
>       - name: Install Go
>         if: matrix.os == 'macos-latest'
> ```

Why is go required on macOS but not Linux?
___
*[`.github/workflows/clippy.yml` line 92 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspZvC7d0UhV3pe2aQ:-NxspZvC7d0UhV3pe2aR:b2nreh6) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L92)):*
> ```YAML
>         uses: actions/checkout@v3
> 
>       - name: Checkout wireguard-go submodule
> ```

Same as above.
___
*[`.github/workflows/desktop-e2e.yml` line 164 at r5](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxlgbXT3fhpjBizNm4M:-NxspvLM4dsFxO0k7rE-:b32v57) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/d5be080ce1d6e6042b29dcfa96f1eb650a0bd070/.github/workflows/desktop-e2e.yml#L164)):*
<details><summary><i>Previously, hulthe (Joakim Hulthe) wrote…</i></summary><blockquote>

better to have it and not need it? :shrug: 
it's probably a non-issue to just keep this
</blockquote></details>

+1
___
*[`.github/workflows/rust-unused-dependencies.yml` line 92 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsqTno52L7jR3zqtBs:-NxsqTno52L7jR3zqtBt:b-l6rasy) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/rust-unused-dependencies.yml#L92)):*
> ```YAML
>         run: cargo udeps --target aarch64-linux-android --package mullvad-jni
> 
>   cargo-udeps-windows:
> ```

The macOS and Windows jobs can probably be merged if just the `Install go` step is skipped on Windows.
___
*[`wireguard-go-rs/libwg/libwg.a` line 0 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxstHwL86AkpHVOlIep:-NxstHwL86AkpHVOlIeq:b1pdtvm) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/wireguard-go-rs/libwg/libwg.a#L0)):*
This should probably be added to `.gitignore`? And completely removed from the git history as well (so it's not needlessly fetched).

**Review by MarkusPettersson98 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)* status: 66 of 77 files reviewed, 6 unresolved discussions (waiting on @dlon, @hulthe, and @Serock3)

___
*[`.github/workflows/android-app.yml` line 126 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsnjvgCWSxm-3nQRzX:-Nxuk2OICZ1DpkY5C_Se:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L126)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/android-app.yml` line 186 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsoG9bBZ2go4Sw0xak:-Nxuk3Ab5gvJdXuAFTN5:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L186)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/android-app.yml` line 245 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsoatN-phMk8DzR9-b:-Nxuk3hL2R0bUXAy_t1S:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L245)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/android-app.yml` line 277 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-Nxsoct55BGIosunA_PN:-Nxuk4E9BE8fbhF1AjCB:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L277)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/android-app.yml` line 386 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsodgL8kfAkVnuLOYg:-Nxuk4lK8ErYzxwcIS4-:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/android-app.yml#L386)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/cargo-vendor.yml` line 24 at r2](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxMYKB0BT0No4ckYtjA:-NxukC9o4TXEzPaxBJ9u:b-jh9ay1) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/359aa96be5e78c84ea4c96c1a6e00b88106aac10/.github/workflows/cargo-vendor.yml#L24)):*
<details><summary><i>Previously, hulthe (Joakim Hulthe) wrote…</i></summary><blockquote>

shouldn't this be wireguard-go-rs/libwg/**wireguard-go**?
</blockquote></details>

The submodules are no longer explicitly stated, so this should be resolved :blush:
___
*[`.github/workflows/cargo-vendor.yml` line 21 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-Nxsoqw03tAOSG_78dIz:-Nxuk5XrB649hKYoF9UG:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/cargo-vendor.yml#L21)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Should we just set `submodules: true` for the `checkout` action instead of fetching this particular submodule? It seems to use `--depth=1` by default, so it shouldn't be too wasteful.
</blockquote></details>

Done
___
*[`.github/workflows/clippy.yml` line 46 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspVf-4Yb-KWHjDED2:-NxukMFI7zrc1DpahU0I:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L46)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Same as above.
</blockquote></details>

Done
___
*[`.github/workflows/clippy.yml` line 62 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspCtq7F_FDkq7Jt63:-NxukNXjA9WzR6Vuktd8:b-f71gdu) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L62)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Why is go required on macOS but not Linux?
</blockquote></details>

Good question. `ubuntu-latest` [includes go](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md), but so does `macos-latest`. I don't really know why it seemed to work on Linux but not macOS :thinking: 

In any case, they do not contain the exact version of go that we want, so I made the `setup-go` step run on Linux as well! :blush:
___
*[`.github/workflows/clippy.yml` line 92 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxspZvC7d0UhV3pe2aQ:-NxulA6R6Nar-0brf-N3:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/clippy.yml#L92)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

Same as above.
</blockquote></details>

Done

**Review by MarkusPettersson98 — COMMENTED** [review]
> Reviewed 1 of 5 files at r9.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231)* status: 61 of 78 files reviewed, 4 unresolved discussions (waiting on @dlon and @hulthe)

___
*[`.github/workflows/desktop-e2e.yml` line 164 at r5](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxlgbXT3fhpjBizNm4M:-NxuuExF7JBA32IEU9oa:b-cstnkg) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/d5be080ce1d6e6042b29dcfa96f1eb650a0bd070/.github/workflows/desktop-e2e.yml#L164)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

+1
</blockquote></details>

I agree
___
*[`.github/workflows/rust-unused-dependencies.yml` line 92 at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxsqTno52L7jR3zqtBs:-NxuwY7m7PiH8BBcmi8v:bb6un1j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/.github/workflows/rust-unused-dependencies.yml#L92)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

The macOS and Windows jobs can probably be merged if just the `Install go` step is skipped on Windows.
</blockquote></details>

Done
___
*[`wireguard-go-rs/libwg/libwg.a` line  at r7](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6231#-NxstHwL86AkpHVOlIep:-NxvCyZ1EusSP_oBJ-Kk:b-rso30f) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/3f1c72e04c13a24fa716db77a19e19766ebb07fc/wireguard-go-rs/libwg/libwg.a#L)):*
<details><summary><i>Previously, dlon (David Lönnhager) wrote…</i></summary><blockquote>

This should probably be added to `.gitignore`? And completely removed from the git history as well (so it's not needlessly fetched).
</blockquote></details>

Done (I think). Please verify that it is actually completely nuked from the git history :blush:

**Pull request merged**